File: app/src/main/java/com/ivanovsky/passnotes/ui/groups/GroupsContract.java
package com.ivanovsky.passnotes.ui.groups;

import com.ivanovsky.passnotes.data.safedb.model.Group;
import com.ivanovsky.passnotes.ui.core.BasePresenter;
import com.ivanovsky.passnotes.ui.core.BaseView;

import java.util.List;

public class GroupsContract {

	interface View extends BaseView<Presenter> {
		void showGroups(List<Group> groups);
		void showNoItems();
		void showNewGroupScreen();
		void showUnlockScreenAndFinish();
		void showNotesScreen(Group group);
	}

	interface Presenter extends BasePresenter {
		void loadData();
		void onGroupClicked(Group group);
	}
}

File: app/src/main/kotlin/com/ivanovsky/passnotes/ui/groups/GroupsContract.kt
package com.ivanovsky.passnotes.ui.groups

import com.ivanovsky.passnotes.data.entity.Group
import com.ivanovsky.passnotes.presentation.core.BasePresenter
import com.ivanovsky.passnotes.presentation.core.BaseView

class GroupsContract {

	interface View: BaseView<Presenter> {
		fun showGroups(groups: List<Group>)
		fun showNoItems()
		fun showNewGroupScreen()
		fun showUnlockScreenAndFinish()
		fun showNotesScreen(group: Group)
	}

	interface Presenter: BasePresenter {
		fun loadData()
		fun onGroupClicked(group: Group)
	}
}